### PR TITLE
🌱 test: add restConfigModifier Option to clusterProxy

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Allows to create a custom ClusterProxy which e.g. adds a custom Dial function to the rest.Config by only overwriting the GetWorkloadCluster function.

xref:
- https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3072

/area e2e-testing

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/framework/cluster_proxy.go [hunk 1]
```diff
@@ -141,6 +141,13 @@ func WithMachineLogCollector(logCollector ClusterLogCollector) Option {
 	}
 }
 
+// WithRESTConfigModifier allows to modify the rest config in GetRESTConfig.
```
Comment (fabriziopandini) [inline test/framework/cluster_proxy.go:144]: nit 
I would like a short explanation about why we need this.
e.g. 
```suggestion
// WithRESTConfigModifier allows to modify the rest config in GetRESTConfig.
// Using this function it is possible to create ClusterProxy that can work with workload clusters hosted in places 
// not directly accessible from the machine where we run the E2E tests, eg. inside kind.
```

**Comment by sbueringer** [comment]
lgtm pending the open comment

**Comment by fabriziopandini** [comment]
/lgtm
/approve
feel free to un hold when tests in CAPV are green

**Pull request merged**